Packages/Status/Sources/Status/Row/Subviews/StatusRowTextView.swift

```diff
@@ -6,17 +6,40 @@ struct StatusRowTextView: View {
   @EnvironmentObject private var theme: Theme
 
   let status: AnyStatus
-  let viewModel: StatusRowViewModel
-
+  @ObservedObject var viewModel: StatusRowViewModel
+  
   var body: some View {
-    HStack {
-      EmojiTextApp(status.content, emojis: status.emojis, language: status.language)
-        .font(.scaledBody)
-        .emojiSize(Font.scaledBodyPointSize)
-        .environment(\.openURL, OpenURLAction { url in
-          viewModel.routerPath.handleStatus(status: status, url: url)
-        })
-      Spacer()
+    VStack {
+      HStack {
+        EmojiTextApp(status.content, emojis: status.emojis, language: status.language, lineLimit: viewModel.collapsedLines)
+          .font(.scaledBody)
+          .emojiSize(Font.scaledBodyPointSize)
+          .environment(\.openURL, OpenURLAction { url in
+            viewModel.routerPath.handleStatus(status: status, url: url)
+          })
+        Spacer()
+      }
+      makeCollapseButton()
+    }
+  }
+  
+  @ViewBuilder
+  func makeCollapseButton() -> some View {
+    if viewModel.showCollapseButton {
+      HStack {
+        Button {
+          withAnimation {
+            viewModel.isCollapsed.toggle()
+          }
+        } label: {
+          Text("Display " + (viewModel.isCollapsed ? "more" : "less"))
```

Comment:
I think it could just be read more and maybe you don't even need to collapse it back?

Reply:
"Read more" is better, yes. I notice when scrolling, that state in the viewmodel is reset once the post is scrolled out a bit and then returned into view, kind of autocollapsing then

Reply:
I would suggest using **Show** more.